tests/plugins/basicstyles/basicstyles.js

```diff
@@ -25,5 +27,203 @@ bender.test( {
 
 			assert.beautified.html( expectedHtml, bot.editor.getData() );
 		} );
+	},
+
+	// (#5215)
+	'test toggle subscript and superscript on selected text': function() {
+		bender.editorBot.create( {
+			name: 'editor-subsup1',
+			config: {
+				coreStyles_disallowSubscriptSuperscript: true
+			}
+		}, function( bot ) {
+			var editor = bot.editor;
+
+			bot.setHtmlWithSelection( '<p>[foo] bar</p>' );
+			editor.execCommand( 'subscript' );
+			assert.areSame(
+				'<p><sub>foo</sub> bar</p>',
+				editor.editable().getData( isNotGeckoOrIe ),
+				'There is no added subscript element' );
+
+			editor.execCommand( 'superscript' );
+			assert.areSame(
+				'<p><sup>foo</sup> bar</p>',
+				editor.editable().getData( isNotGeckoOrIe ),
+				'There both subscript and superscript elements' );
+		} );
+	},
+
+	// (#5215)
+	'test properly toggle subscript and superscript on selected text with other basic styles': function() {
+		bender.editorBot.create( {
+			name: 'editor-subsup2',
+			config: {
+				coreStyles_disallowSubscriptSuperscript: true
+			}
+		}, function( bot ) {
+			var editor = bot.editor;
+
+			bot.setHtmlWithSelection( '<p>[foo] bar</p>' );
+			editor.execCommand( 'bold' );
```

Comment:
It was on purpose to check if toggling does not change subscript and superscript behavior. I'll adjiust this test but IMO it would be good to leave the rest tests that toggle sub and sup as it now, but if you feel that it's unnecessary I can adjust it.